spi/spi_master_slave/spi_slave/spi_slave.c

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2021 Michael Stoops. All rights reserved.
+ * Portions copyright (c) 2021 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+// Example of an SPI bus slave using the PL022 SPI interface
+
+#include <stdio.h>
+#include <string.h>
+#include "pico/stdlib.h"
+#include "pico/binary_info.h"
+#include "hardware/spi.h"
+
+#define BUF_LEN         0x100
+
+void printbuf(uint8_t buf[], size_t len) {
+    int i;
+    for (i = 0; i < len; ++i) {
+        if (i % 16 == 15)
+            printf("%02x\n", buf[i]);
+        else
+            printf("%02x ", buf[i]);
+    }
+
+    // append trailing newline if there isn't one
+    if (i % 16) {
+        putchar('\n');
+    }
+}
+
+
+int main() {
+    // Enable UART so we can print
+    stdio_init_all();
+#if !defined(spi_default) || !defined(PICO_DEFAULT_SPI_SCK_PIN) || !defined(PICO_DEFAULT_SPI_TX_PIN) || !defined(PICO_DEFAULT_SPI_RX_PIN) || !defined(PICO_DEFAULT_SPI_CSN_PIN)
+#warning spi/spi_slave example requires a board with SPI pins
+    puts("Default SPI pins were not defined");
+#else
+
+    printf("SPI slave example\n");
+
+    // Enable SPI 0 at 1 KHz and connect to GPIOs
+    spi_init(spi_default, 1000);
+    spi_set_slave(spi_default, true);
+    gpio_set_function(PICO_DEFAULT_SPI_RX_PIN, GPIO_FUNC_SPI);
+    gpio_set_function(PICO_DEFAULT_SPI_SCK_PIN, GPIO_FUNC_SPI);
+    gpio_set_function(PICO_DEFAULT_SPI_TX_PIN, GPIO_FUNC_SPI);
+    gpio_set_function(PICO_DEFAULT_SPI_CSN_PIN, GPIO_FUNC_SPI);
+    // Make the SPI pins available to picotool
+    bi_decl(bi_4pins_with_func(PICO_DEFAULT_SPI_RX_PIN, PICO_DEFAULT_SPI_TX_PIN, PICO_DEFAULT_SPI_SCK_PIN, PICO_DEFAULT_SPI_CSN_PIN, GPIO_FUNC_SPI));
+
+    printf("SPI slave initialised.\n");
+
+    uint8_t out_buf[BUF_LEN], in_buf[BUF_LEN];
+
+    // Initialize output buffer
+    for (size_t i = 0; i < BUF_LEN; ++i) {
+        // bit-inverted from i. The values should be: {0xff, 0xfe, 0xfd...}
+        out_buf[i] = ~i;
+    }
+
+    printf("When reading from MOSI, the following buffer will be written to MISO:\n");
+    printbuf(out_buf, BUF_LEN);
+    
+    for (size_t i = 0; true; ++i) {
```

Comment:
I guess this could be a while true loop?

Reply:
But still needs the initializer and increment, so I don't think that's a net gain. I think I'll leave it.

Reply:
Nit: You can delete the "true".

Reply:
> Nit: You can delete the "true".

Nice. Thanks!

Reply:
Changed in next rev. Resolving.

Reply:
When I originally added this comment, `i` wasn't used in this slave-loop, but it is now :wink:

Reply:
Ohhhhh... Got it. My bad.